scripts/pre_commit_hook.py

```diff
@@ -0,0 +1,132 @@
+but it will have no effect.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+
+def _install_hook():
+    """Installs the pre_commit_hook script. It ensures that oppia is root."""
+    oppia_dir = os.getcwd()
+    hooks_dir = os.path.join(oppia_dir, '.git', 'hooks')
+    pre_commit_file = os.path.join(hooks_dir, 'pre-commit')
+    if os.path.islink(pre_commit_file):
+        print 'Symlink already exists'
+        return
+    try:
+        os.symlink(os.path.abspath(__file__), pre_commit_file)
+        print 'Created symlink in .git/hooks directory'
+    # Raises AttributeError on windows, OSError added as failsafe.
+    except (OSError, AttributeError):
+        shutil.copy(__file__, pre_commit_file)
+        print 'Copied file to .git/hooks directory'
+
+
+def _start_subprocess_for_result(cmd):
+    """Starts subprocess and returns (stdout, stderr)."""
+    task = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+    out, err = task.communicate()
+    return out, err
+
+
+def _does_diff_include_package_lock_file_and_no_package_file():
+    """Checks whether the diff includes package-lock.json and
+        no package.json.
+
+    Returns:
+        bool. Whether the diff includes package-lock.json and
+            no package.json.
+
+    Raises:
+        ValueError if git command fails.
+    """
+
+    git_cmd = ['git', 'diff', '--name-only', '--cached']
+    out, err = _start_subprocess_for_result(git_cmd)
+
+    if not err:
+        files_changed = out.split('\n')
+        return 'package-lock.json' in files_changed and (
+            'package.json' not in files_changed)
+    else:
+        raise ValueError(err)
+
+
+def _revert_changes_in_package_lock_file():
+    """Reverts the changes made in package-lock.json file.
+
+    Raises:
+        ValueError if any git command fails.
+    """
+
+    git_unstage_cmd = ['git', 'reset', '--', 'package-lock.json']
+    git_revert_cmd = ['git', 'checkout', '--', 'package-lock.json']
+
+    _, err_unstage_cmd = (
+        _start_subprocess_for_result(git_unstage_cmd))
+
+    if not err_unstage_cmd:
+        _, err_revert_cmd = (
+            _start_subprocess_for_result(git_revert_cmd))
+
+        if not err_revert_cmd:
+            print 'Changes to package-lock.json successfully reverted!'
+        else:
+            raise ValueError(err_revert_cmd)
+    else:
+        raise ValueError(err_unstage_cmd)
+
+
+def main():
+    """Main method for pre-commit hook that checks files added/modified
+    in a commit.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--install', action='store_true', default=False,
+                        help='Install pre_commit_hook to the .git/hooks dir')
+    args = parser.parse_args()
+    if args.install:
+        _install_hook()
+        sys.exit(0)
+
+    print 'Running pre-commit check for package-lock.json ..'
```

Comment:
Here and below: typically, use three dots for ellipsis.

Reply:
Done, thanks!